modules/command-engine/core/src/main/scala/surge/internal/core/SurgePartitionRouterImpl.scala

```diff
@@ -60,11 +66,17 @@ private[surge] final class SurgePartitionRouterImpl(
       businessLogic.partitioner,
       businessLogic.kafka.stateTopic,
       regionCreator,
-      RoutableMessage.extractEntityId)(businessLogic.tracer)
+      RoutableMessage.extractEntityId,
+      kafkaProducerOverride)(businessLogic.tracer)
 
     val routerActorName = s"${businessLogic.aggregateName}RouterActor"
-    val shardRouter = system.actorOf(shardRouterProps, name = routerActorName)
-    shardRouter
+    val shardRouterManaged = ActorLifecycleManagerActor.manage(
+      actorSystem = system,
+      managedActorProps = shardRouterProps,
+      componentName = routerActorName,
+      actorLifecycleName = Some(routerActorName),
```

Comment:
Actually one thing you could do to validate there are sufficient tests is remove the `actorLifecycleName` parameter (or pass in a `None`) and validate that we have a failing test that was the bug around this. Having a reliably failing test around that would ease my anxiety about re-wrapping this in a managed actor